SDWebImage/SDImageCache.h

```diff
@@ -54,12 +54,32 @@ typedef void(^SDWebImageCompletionWithPossibleErrorBlock)(NSError * _Nullable er
 #pragma mark - Properties
 
 /**
- *  Cache Config object - storing all kind of settings
+ *  Cache Config object - storing all kind of settings.
+ *  The property is copy so change of currrent config will not accidentally affect other cache's config.
  */
-@property (nonatomic, nonnull, readonly) SDImageCacheConfig *config;
+@property (nonatomic, copy, nonnull, readonly) SDImageCacheConfig *config;
+
+/**
+ *  The disk cache's root path
+ */
+@property (nonatomic, copy, nonnull, readonly) NSString *diskCachePath;
+
+/**
+ *  The additional disk cache path to check if the query from disk cache not exist;
+ *  The `key` param is the image cache key. You should return the specify file path to load image from disk. You can use built-in hash method(See `cacheFileNameForKey:` in SDDiskCache) or your own way to provide mapping between cache key and file path.
+ *  Useful if you want to bundle pre-loaded images with your app
+ */
+@property (nonatomic, copy, nullable) SDImageCacheAdditionalCachePathBlock additionalCachePathBlock;
 
 #pragma mark - Singleton and initialization
 
+/**
+ Set the default cache config used for shared instance or initialization which does not provide any cache config
+
+ @param config The default cache config to use. You should not pass nil.
+ */
++ (void)setDefaultCacheConfig:(nonnull SDImageCacheConfig *)config;
```

Comment:
There should be a class level default config, or we have no timing to change the custom memory class or disk cache for the `sharedCache`.